components/brave_wallet/browser/blockchain_registry.cc

```diff
@@ -100,26 +100,55 @@ void BlockchainRegistry::GetAllTokens(const std::string& chain_id,
   std::move(callback).Run(std::move(tokens_copy));
 }
 
-void BlockchainRegistry::GetBuyTokens(const std::string& chain_id,
+void BlockchainRegistry::GetBuyTokens(mojom::OnRampProvider provider,
+                                      const std::string& chain_id,
                                       GetBuyTokensCallback callback) {
   std::vector<brave_wallet::mojom::BlockchainTokenPtr> blockchain_buy_tokens;
-  for (auto token : *kBuyTokens) {
+  std::vector<mojom::BlockchainToken>* buy_tokens = nullptr;
+  if (provider == mojom::OnRampProvider::Wyre)
+    buy_tokens = &GetWyreBuyTokens();
+  else if (provider == mojom::OnRampProvider::Ramp)
+    buy_tokens = &GetRampBuyTokens();
+
+  for (const auto& token : *buy_tokens) {
```

Comment:
Please also check if buy_tokens pointer is not null before iterating.

Reply:
Fixed 389ca5a6e0d73c51210b249a36e0bcc4b394878b.